src/Microsoft.AspNet.OData/Formatter/ODataMediaTypeFormatter.cs

```diff
@@ -117,7 +117,10 @@ public override MediaTypeFormatter GetPerRequestFormatterInstance(Type type, Htt
             else
             {
                 ODataVersion version = GetODataResponseVersion(request);
-                return new ODataMediaTypeFormatter(this, version, request);
+                return new ODataMediaTypeFormatter(this, version, request)
+                {
+                    BaseAddressFactory = this.BaseAddressFactory
```

Comment:
Should we let the copy cstor set this property instead of explicitly setting it here, so that when we add other copy cstor calls this is automatically included?